h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -431,7 +437,9 @@ public void compute2() {
             rs._tsk.join();
           } else if(rs.isMutating()) // c) => throw IAE
             throw new IllegalArgumentException("Can not compute rollup stats while vec is being modified. (3)");
-        } else { // d) => compute the rollups
+        } else { // rs == null d) => compute the rollups
+          final Vec vec = DKV.getGet(_vecKey);
+          if (vec == null) break; // Vec was deleted since we last checked
```

Comment:
I just realized there is one scenario that is not covered:

The rollups are computed with existing Vec and no existing rollups. Then the rollups are null before and after the delete (rs = null), it doesn't matter that we marked the Vec a mutating because the we can delete the rollups with mutating flag before we actually try to install them again.

1. We find out there are no rollups (rs == null)
2. We calculate the rollups
3. The vec is set as mutating and the delete completes (the rollups are again deleted rs == null)
4. We install the new rollups (no problem here, it seems like nothing changed)